src/main/java/com/test/scrapers/JsoupScraper.java

```diff
@@ -0,0 +1,163 @@
+package com.test.scrapers;
+
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.jsoup.select.Elements;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.openqa.selenium.WebDriver;
+import com.machinepublishers.jbrowserdriver.Timezone;
+import com.machinepublishers.jbrowserdriver.JBrowserDriver;
+import com.machinepublishers.jbrowserdriver.Settings;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.HashSet;
+import java.util.Properties;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Web scraper that uses JSoup (http://jsoup.org)
+ */
+public class JsoupScraper implements Scraper {
+
+    private static final Logger logger = LoggerFactory.getLogger(JsoupScraper.class);
+
+    private final Set<String> visitedUrls = new HashSet<>();
+    private final Set<String> emails = new HashSet<>();
+    private final Pattern emailRegEx = Pattern.compile("[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\\.[a-zA-Z0-9-.]+");
+    private final boolean useWebClient;
+
+    public JsoupScraper(boolean useWebClient) {
+        this.useWebClient = useWebClient;
+    }
+
+    public Set<String> getEmails(String url) throws Exception {
+        if (!url.contains("://")) {
+            url = "http://" + url;
+        }
+        URI rootUri = null;
+        try {
+            rootUri = new URI(url);
+        } catch (URISyntaxException e) {
+            System.err.println("Invalid URL: " + url);
+            System.exit(-1);
+        }
+
+        try {
+            scrapeEmailsFromPage(rootUri, url);
+        } catch (Exception ex) {
+            if (emails.size() == 0) {
+                throw ex;
+            } else {
+                // just log a debug warning because we got some useful results back
+                logger.warn("Could not finish scraping site: " + ex);
+            }
+        }
+
+        return emails;
+    }
+
+    private void scrapeEmailsFromPage(URI rootUri, String path) throws Exception {
+        if (path.startsWith("javascript:") || path.startsWith("#")) {
+            return;
+        }
+        if (path.startsWith("mailto:")) {
+            String email = path.substring("mailto".length()+1);
+            emails.add(email);
+            return;
+        }
+
+        String url;
+        if (path.contains("://")) {
+            URI pathUri = new URI(path);
+            if (!pathUri.getHost().contains(rootUri.getHost())) {
```

Comment:
When I ran this on "jana.com," the scraper ended up on developer.android.com and www.genymotion.com, and was still running after about 30 minutes.  What would you do to solve for that?